Formula/nginx-unit.rb

```diff
@@ -0,0 +1,256 @@
+class NginxUnit < Formula
+  desc "Dynamic web and application server for multiple languages"
+  homepage "https://unit.nginx.org/"
+  url "https://unit.nginx.org/download/unit-1.1.tar.gz"
+  sha256 "7c66365f5ea87e7e8903bac9d639676947fe2ab7aa799ce35defc426b3409ee0"
+
+  depends_on "openssl"
+  depends_on "pcre"
+
+  def install
+    openssl = Formula["openssl"]
+    pcre = Formula["pcre"]
+
+    system "./configure",
+      "--prefix=#{prefix}",
+      "--bindir=#{bin}",
+      "--sbindir=#{bin}",
+      # This tells the unit daemon where it can load modules from. Configuring
+      # the location at runtime is painful. We want this to be in the
+      # $HOMEBREW_PREFIX so that other formulae can link in additional modules.
+      "--modules=#{HOMEBREW_PREFIX}/lib/unit",
+      "--state=#{var}/unit",
+      "--log=#{var}/log/unit.log",
+      "--pid=#{var}/run/unit.pid",
+      "--control=unix:#{var}/run/unit.control.sock",
+      "--cc-opt=-I#{pcre.opt_include} -I#{openssl.opt_include}",
+      "--ld-opt=-L#{pcre.opt_lib} -L#{openssl.opt_lib}"
+
+    system "make", "install"
+
+    # Beyond here we want to install modules into the keg, so we'll move them
+    # into the prefix ourselves
+    libunit = lib/"unit"
+    libunit.mkpath
+
+    system "./configure", "perl"
+    system "make", "perl"
+    libunit.install "build/perl.unit.so"
+
+    system "./configure", "python"
+    system "make", "python"
+    libunit.install "build/python.unit.so"
+
+    system "./configure", "ruby"
+    system "make", "ruby"
+    libunit.install "build/ruby.unit.so"
+  end
+
+  def post_install
+    (var/"run").mkpath
+  end
+
+  def caveats
+    <<~EOS
+      Once running, you can control unit using the control socket, for example using
+      curl:
+        curl --unix-socket #{var}/run/unit.control.sock http://localhost/
+
+      Modules are available for system perl, python and ruby. The system php doesn't
```

Comment:
The bit about php seems irrelevant.

Reply:
Folks installing unit might expect it to be able to run php apps using the macOS-installed php. It seems an important caveat?

Reply:
Can we build the php module if we use Homebrew's php?

Reply:
I tried, it's difficult. This much seemed like a good first step, building against everything available on the system without adding dependencies. I was going to propose an `nginx-unit-php` formula separately which builds just the php shared module against homebrew php and puts in into the `/usr/local/lib/unit` directory. Potentially also `nginx-unit-php@5.6` etc, as many versions can be used simultaneously — part of the pitch of unit. This could be done for other homebrew formulae with interop, like homebrew' perl, python, ruby, and go.

Reply:
I see. In any case, I think the php reference should be removed since the fact that it doesn't work with macOS's php isn't specific to Homebrew and the caveats are for Homebrew specific information and path locations only.